crates/ursa/src/ursa/mod.rs

```diff
@@ -55,38 +55,22 @@ pub struct CliOpts {
 
 impl CliOpts {
     pub fn to_config(&self) -> Result<UrsaConfig> {
-        let mut path = PathBuf::from(env!("HOME")).join(DEFAULT_CONFIG_PATH_STR);
-        if let Some(config_file) = &self.config {
-            info!(
-                "Reading configuration from user provided config file {}",
-                config_file
-            );
-            path = PathBuf::from(&config_file);
+        let mut config = load_config(
+            &self
+                .config
+                .as_ref()
+                .map(|p| PathBuf::from(p).resolve().to_path_buf())
+                .unwrap_or_else(|| PathBuf::from(var("HOME").unwrap_or_default()).join(DEFAULT_CONFIG_PATH_STR)),
```

Comment:
or you can just use [this crate](https://crates.io/crates/home) and be safe.

Reply:
used `dirs::home_dir` which `resolve_path` uses internally. It has 34 million crate downloads so should be pretty safe